common/djangoapps/credo_modules/middleware.py

```diff
@@ -23,7 +24,8 @@ def process_response(self, request, response):
 class CheckCredoAdditionalProfile(object):
 
     def process_request(self, request):
-        if not request.path.startswith('/credo_modules/profile') and request.user.is_authenticated():
+        if not request.path.startswith('/credo_modules/profile') and not request.path.startswith(
+                '/lti_provider') and request.user.is_authenticated() and self.is_graded_block(request):
```

Comment:
`is_graded_block` this check should be used inside function `user_must_fill_additional_profile_fields` because function `user_must_fill_additional_profile_fields` used twice: here in the current middleware and here: https://github.com/CredoReference/edx-platform/blob/integration/lms/djangoapps/courseware/views.py#L1388